products.php

```diff
@@ -0,0 +1,51 @@
+<?php
+require_once 'src/Models/CategoryModel.php';
+require_once 'src/Services/ProductsDisplayService.php';
+require_once 'src/Models/ProductModel.php';
+require_once 'src/Factory/furnitureDatabaseConnector.php';
+require_once 'src/Entities/ProductsEntity.php';
+
+$db = furnitureDatabaseConnector::connect();
+define('NUM_OF_CATEGORIES', 11);
+$id = $_GET["id"];
+if (!isset($id) || !is_numeric($id) || $id > NUM_OF_CATEGORIES || $id < 1) {
+    echo '<h1>Invalid id number</h1>
```

Comment:
Vague error messages like 'oops! something went wrong' are generally preferred. If someone has nefarious intent we don't want to give them any useful information when they try to break our application.